### ISSUE TITLE
OpenSWATH support for peptide internal fragment ions

### ISSUE THREAD

**Issue by karlssoc** [OpenMS/OpenMS#4373]
Hi,

Peptide internal fragment ions seem to occur frequently among the most intense signals from fragment spectra generated by HCD fragmentation (https://pubs.acs.org/doi/10.1021/acs.jproteome.6b00928), denoted as m ions. 

OpenSWATH does currently not have support m fragment ions for assay generation, decoy generation and  data extraction. This can currently be worked around by using other tools for assay library creation and forcing data extraction, but it would be nice with native support.

Thanks,
Christofer

**Comment by hroest** [comment]
I agree, I would say the main limitations are assay generation and decoy generation. OpenSWATH extraction and scoring actually does not really care that much what your ions represent. I wonder though how strong of an improvement you would expect in peptide id when adding these m ions. Have you seen that in your data? Do you have an example for an input for the AssayGenerator with such ions so that we could add support (or help us to implement support)?
